### PR TITLE
v0.5.0

### PR DESCRIPTION
**Pull request by eschultink**
v0.5.0 back to main

Next steps, after that's merged to `main`:
  1. publish the release: `./tools/release/publish.sh v0.5.0`
  2. update stable deployment in demos repo to point to v0.5.0
  3. prep next rc: `./tools/release/prep.sh v0.5.0 rc-v{x.y.z}`

### CODE REVIEW

File: docs/faq-security.md [hunk 1]
```diff
@@ -9,17 +9,13 @@ to perform santization generally across many fields and endpoints.
...
+Worklytics tenant. Please contact [sales@worklytics.co](mailto:sales@worklytics.co] to access this
+feature.
 
-If you take this approach, you will be responsible for updating your IP restrictions frequently as
-GCP changes their IP blocks, or your data flow to Worklytics may break. As such, this is not
-officially supported by Worklytics. For an example of how to do this, see [worklytics-ip-blocks](../infra/modules/worklytics-ip-blocks/README.md) module.
-
-Your Worklytics tenant is a process running in GCP, personified by a unique GCP service account. You
-simply use your cloud's IAM to grant that service account access to your psoxy instance.
+Note that all requests from your Worklytics tenant to your Psoxy instances are authenticated via identity
```
Comment (davidfq) [inline docs/faq-security.md:16]:
```suggestion
Note that all requests from your Worklytics tenant to your Pseudonymizing Proxy instances are authenticated via identity
```

File: docs/faq-security.md [hunk 2]
```diff
@@ -9,17 +9,13 @@ to perform santization generally across many fields and endpoints.
 
 ## Can Psoxy invocation be locked to a set of known IP addresses?
```
Comment (davidfq) [inline docs/faq-security.md:10]:
```suggestion
## Can Worklytics Pseudonymizing Proxy invocation be locked to a set of known IP addresses?
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,25 @@ Changes to be including in future/planned release notes will be added here.
...
+    portal (formally Azure AD portal blade)
+  - variables to `aws-host`/`gcp-host` modules to have changed slightly; if you initially copied an
+    example based on 0.4.x, you may have to update some variable names in your `main.tf`.
+  - min `google` provider is now 5.0; this applies whether you're using GCP-hosted proxy, or merely Google Workspace as a
+    data source
+  - various migrations applicable to 0.4.x have been removed; if upgrading from 0.4.x, make sure you first upgrade to latest version of 0.4.x (eg, 0.4.61), run
+    `terraform apply`, and THEN update to 0.5.x
+  - the v0.3 pseudonymization algorithm is no longer supported; attempting to do so should result in an error
+  - `scope` field will no longer be sent with JSON-encoded pseudonyms.
+  - min java version in now 17; java 11 no longer supported (as it's a deprecated runtime in GCP; and Oracle supported has ended)
```
Comment (jlorper) [inline CHANGELOG.md:27]:
```suggestion
  - minimum `azuread` provider version is generally 2.44; if you're using an older version, you'll need to
    upgrade (`terraform init --upgrade`); a state refresh (`terraform refresh`) may help if it complains about unknown attributes
    present in your state
   - `azuread-local-cert` module variables have changed; you must now pass `application_id` instead
    of `application_object_id`; these refer to different values you can obtain via the [Microsoft Entra admin center](https://entra.microsoft.com/#home)
    portal (formally Azure AD portal blade)
  - variables to `aws-host`/`gcp-host` modules to have changed slightly; if you initially copied an
    example based on 0.4.x, you may have to update some variable names in your `main.tf`.
  - minimum `google` provider version is now 5.0; this applies whether you're using GCP-hosted proxy, or merely Google Workspace as a
    data source
  - various migrations applicable to 0.4.x have been removed; if upgrading from 0.4.x, make sure you first upgrade to latest version of 0.4.x (eg, 0.4.61), run
    `terraform apply`, and THEN update to 0.5.x
  - the v0.3 pseudonymization algorithm is no longer supported; attempting to do so should result in an error
  - `scope` field will no longer be sent with JSON-encoded pseudonyms.
  - minimum java version in now 17; java 11 no longer supported (as it's a deprecated runtime in GCP; and Oracle support has ended)
```

**Review by jlorper — COMMENTED** [review]
Looks good, just added one suggestion on the readme

**Review by jlorper — APPROVED** [review]

**Pull request merged**